library/data/loot_table/src/main/java/org/quiltmc/qsl/loot/api/QuiltLootPoolBuilder.java

```diff
@@ -0,0 +1,176 @@
+import net.minecraft.loot.provider.number.LootNumberProvider;
+
+import org.quiltmc.qsl.loot.mixin.LootPoolBuilderAccessor;
+
+/**
+ * Quilt's version of {@link net.minecraft.loot.LootPool.Builder}. Adds additional methods and
+ * hooks not found in the original class.
+ *
+ * <p>To create a new instance of this class, use {@link #builder()}.
+ *
+ * @see #copyOf(LootPool)
+ */
+public class QuiltLootPoolBuilder extends LootPool.Builder {
+	private final LootPoolBuilderAccessor access = (LootPoolBuilderAccessor) this;
+
+	private QuiltLootPoolBuilder() {
+	}
+
+	// Vanilla overrides
+
+	@Override
+	public QuiltLootPoolBuilder rolls(LootNumberProvider rolls) {
+		super.rolls(rolls);
+		return this;
+	}
+
+	@Override
+	public QuiltLootPoolBuilder with(LootPoolEntry.Builder<?> entry) {
+		super.with(entry);
+		return this;
+	}
+
+	@Override
+	public QuiltLootPoolBuilder conditionally(LootCondition.Builder builder) {
+		super.conditionally(builder);
+		return this;
+	}
+
+	@Override
+	public QuiltLootPoolBuilder apply(LootFunction.Builder builder) {
+		super.apply(builder);
+		return this;
+	}
+
+	// Additional methods
+
+	/**
+	 * Adds an entry to this builder.
+	 *
+	 * @param entry the entry to add
+	 * @return this builder
+	 */
+	public QuiltLootPoolBuilder with(LootPoolEntry entry) {
+		access.getEntries().add(entry);
+		return this;
+	}
+
+	/**
+	 * Adds entries to this builder.
+	 *
+	 * @param entries the entries to add
+	 * @return this builder
+	 */
+	public QuiltLootPoolBuilder with(Collection<? extends LootPoolEntry> entries) {
+		access.getEntries().addAll(entries);
+		return this;
+	}
+
+	/**
+	 * Adds a condition to this builder.
+	 *
+	 * @param condition the condition to add
+	 * @return this builder
+	 */
+	public QuiltLootPoolBuilder conditionally(LootCondition condition) {
+		access.getConditions().add(condition);
+		return this;
+	}
+
+	/**
+	 * Adds conditions to this builder.
+	 *
+	 * @param conditions the conditions to add
+	 * @return this builder
+	 */
+	public QuiltLootPoolBuilder conditionally(Collection<? extends LootCondition> conditions) {
+		access.getConditions().addAll(conditions);
+		return this;
+	}
+
+	/**
+	 * Applies a function to this builder.
+	 *
+	 * @param function the function to apply
+	 * @return this builder
+	 */
+	public QuiltLootPoolBuilder apply(LootFunction function) {
+		access.getFunctions().add(function);
+		return this;
+	}
+
+	/**
+	 * Applies functions to this builder.
+	 *
+	 * @param functions the functions to apply
+	 * @return this builder
+	 */
+	public QuiltLootPoolBuilder apply(Collection<? extends LootFunction> functions) {
+		access.getFunctions().addAll(functions);
+		return this;
+	}
+
+	/**
+	 * Sets the bonus rolls of this builder.
+	 *
+	 * @param bonusRolls the bonus rolls range
+	 * @return this builder
+	 * @see QuiltLootPools#getBonusRolls(LootPool)
+	 */
+	public QuiltLootPoolBuilder bonusRolls(LootNumberProvider bonusRolls) {
+		access.setBonusRollsRange(bonusRolls);
```

Comment:
This can be replaced with a super-calling `@Override` method (this was added to `LootPool.Builder` in 1.16)